test-cypress/integration/explore/visual.graph.spec.js

```diff
@@ -1,357 +1,203 @@
-                    cy.get('div.filter-sidepanel.tab-content.ng-scope > div > div.form-group.mb-2 > div.checkbox' +
-                        '> label > input[ng-model="settings[\'includeInferred\']"]')
-                        .check();
-                } else {
-                    // Include inferred: false
-                    cy.get('div.filter-sidepanel.tab-content.ng-scope > div > div.form-group.mb-2 > div.checkbox' +
-                        '> label > input[ng-model="settings[\'includeInferred\']"]')
-                        .uncheck();
-                }
-                cy.get('button[type="submit"]').click();
-            });
-    };
-
-    before(function () {
-        cy.visit('/repository');
-        // Create new one
-        cy.createNewRepo(repoId).wait(2000);
-        cy.setRepoDefault(repoId).wait(500);
-        cy.selectRepo(repoId);
-
-        // Import Wine dataset which will be used in most of the following tests
-        cy.navigateToPage('Import', 'RDF');
-        cy.openImportURLDialog('https://www.w3.org/TR/owl-guide/wine.rdf');
-        cy.clickImportUrlBtn();
-        cy.clickImportBtnOnPopUpMenu().wait(2000);
+describe('Visual graph screen validation', () => {
+
+    let repositoryId = 'graphRepo' + Date.now();
+    const VALID_RESOURCE = 'USRegion';
+
+    beforeEach(() => {
+        repositoryId = 'repo' + Date.now();
+        cy.createRepository({id: repositoryId});
+        cy.enableAutocomplete(repositoryId);
+        cy.importFromUrl(repositoryId, 'https://www.w3.org/TR/owl-guide/wine.rdf', {});
+        cy.presetRepositoryCookie(repositoryId);
+        cy.visit('/graphs-visualizations');
     });
 
-    beforeEach(function () {
-        // Go to Visual graph page
-        cy.visit('/graphs-visualizations').wait(1000);
+    afterEach(() => {
+        cy.deleteRepository(repositoryId);
     });
 
-    after(function () {
-        cy.navigateToPage('Setup', 'Repositories');
-        cy.deleteRepo(repoId);
-    });
-
-    it('Test autocomplete enable toast message', function () {
-        // Search for a resource
-        cy.get('div.ng-scope > :nth-child(1) > .card-block > .ng-isolate-scope > .input-group > .form-control').type('http://');
+    it('Test notification when autocomplete is disabled', () => {
+        // Disable it for the test only
+        cy.disableAutocomplete(repositoryId);
+        getSearchField().type('http://');
         // Verify that a message with a redirection to the autocomplete section is displayed.
-        cy.get('div[class="autocomplete-toast"] > a[href="autocomplete"]')
-            .then(($el) => {
-                expect($el.text()).to.contain('Autocomplete is OFF');
-                expect($el.text()).to.contain('Go to Setup -> Autocomplete');
-            });
-
-        // Enable autocomplete
-        cy.navigateToPage('Setup', 'Autocomplete');
-        cy.get('label[for="toggleIndex"]').click().wait(2000);
+        cy.get('.autocomplete-toast a').should(($link) => {
```

Comment:
Or `should('contain', 'Autocomplete is OFF').and('contain', 'Go to Setup -> Autocomplete').click()` ?

Reply:
yeah nice, I was wondering if there is a more elegant way to do it also without the ugly callback